src/graph/traversal.py

```
@@ -0,0 +1,108 @@
+    queue = deque([(source, [source])])
+    visited = {source}
+
+    while queue:
+        node, path = queue.popleft()
+        if len(path) > max_depth:
+            continue
+        if node == target:
+            paths.append(path)
+            continue
+        for neighbor in adj.get(node, []):
+            new_path = path + [neighbor]
```

Ruff (0.15.18)

[warning] Consider `[*path, neighbor]` instead of concatenation

Replace with `[*path, neighbor]`

(RUF005)